service/books/book_handler.go

```diff
@@ -0,0 +1,70 @@
+package books
+
+import (
+	"fmt"
+	"gin/config"
+	"gin/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	"net/http"
+)
+
+func GetBook(c *gin.Context) {
+	id := c.Param("id")
+	book, err := getBook(config.ElasticClient, id)
```

Comment:
What if we change the book store DB to not use ElasticSearch? Let's try and write it more generic.
Use an interface for the book store operations and have the handlers file not know the underlying implementation.
Same for the Redis cache